doc/source/serve/direct-ingress.md

```diff
@@ -0,0 +1,101 @@
+# Direct Ingress
+
+In the 2.1, Serve provides alpha version gRPC ingress. In this section, you will learn how to 
+
+* use Serve internal gRPC schema to send traffic
+* bring your own gRPC schema into serve application
+
+## Use Serve Schema
+
+Internally, serve provides a simple gRPC schema.
+```
+message PredictRequest {
+  map<string, bytes> input = 2;
+}
+
+message PredictResponse {
+  bytes prediction = 1;
+}
+
+service PredictAPIsService {
+  rpc Predict(PredictRequest) returns (PredictResponse);
+}
+```
+
+Code example:
+
+Server
+```{literalinclude} ../serve/doc_code/direct_ingress.py
+:start-after: __begin_server__
+:end-before: __end_server__
+:language: python
+```
+
+Client
+```{literalinclude} ../serve/doc_code/direct_ingress.py
+:start-after: __begin_client__
+:end-before: __end_client__
+:language: python
+```
+
+:::{note}
+* `input` is a dictionary of `map<string, bytes> ` following the schema described
+*  User input data needs to be serialized to bytes type and feed into the `input`.
+* Response will be under bytes type, which means user code is responsible for to serialize the output into bytes. 
+* By default, the gRPC port is 9000.
+:::
+
+### Client Schema code generation
+There are lots of ways to generate client schema code. Here is a simple step to generate the code.
```

Comment:
Can you add a comment about when you need to do this? Do you always need to do it?  The user might be a little surprised because usually Ray features work out of the box without extra installation, so it might be worth adding a comment here

Reply:
When using RPC protocol, it is common that the user needs to have client code on the client side to construct the corresponding request which is outside Ray Serve scope.

Let's keep it as it is, if we receive lots of questions about it, and then let's update it. At least, it is a very common thing to me :)

Reply:
Sounds good!